src/Common/src/System/Diagnostics/Debug.Unix.cs

```diff
@@ -8,108 +8,118 @@ namespace System.Diagnostics
 {
     // Intentionally excluding visibility so it defaults to internal except for
     // the one public version in System.Diagnostics.Debug which defines
-    // another version of this partial class with the public visibility 
+    // another version of this partial class with the public visibility
     static partial class Debug
     {
-        private static string NewLine => "\n"; 
+#if !MONO
```

Comment:
Sorry I think I got something mixed in my head (condition scrolled off the page), as I missed that both were being included. 

In that case I agree I don't see a great way to do a clean fork purely at the file level without #ifdef'ing the code that doesn't require duplication of some sort. We could create 2 more files that does the additional forking but I'm not sure it is worth it, so I'm OK with your current approach (although we might take another look when similar patterns emerge).